### ISSUE TITLE
[macOS] QGIS 3.18 (GDAL3.2.1) raises sqlite3.OperationalError: disk I/O error

### ISSUE THREAD

**Issue by PeterPetrik** [lutraconsulting/qgis-mergin-plugin#225]
1. open mergin project
2. digitize feature
3. save project and click on "mergin project info"
4. digitize feature & save (here it sometimes hides all the geopackage layers from canvas)
5. click on "mergin project info" -> `sqlite3.OperationalError: disk I/O error`

looks like the GDAL/OGR QGIS connection is interrupted in point 4

**Comment by wonder-sk** [comment]
This error seems linked to warnings in client logs:
```
GEODIFF: SQLITE3: (283)recovered X frames from WAL file /Users/path/path/filename.gpkg-wal
```
which can happen right after python client does checkpoint+vacuum in `do_sqlite_checkpoint()`

Out of many diagnostic log files, this warning only ever appears in logs from Mac users (never windows / linux, or from Input).

Random thougths:
- is sqlite3 on mac compiled as thread safe? On my linux `pragma COMPILE_OPTIONS` in sqlite3 command line reports `THREADSAFE=1` among other things. 
- VACUUM looks like quite a brutal command, rebuilding the whole database. This post has a lot of interesting content https://blogs.gnome.org/jnelson/2015/01/06/sqlite-vacuum-and-auto_vacuum/
   If we need it, maybe better to use VACUUM INTO variant which would not modify the original DB - https://www.sqlite.org/lang_vacuum.html
- instead of checkpoint + vacuum, we may want to use sqlite backup API to safely create a copy of the database - https://www.sqlite.org/backup.html

**Comment by wonder-sk** [comment]
Likely related to https://www.sqlite.org/howtocorrupt.html  - "
2.2.1. Multiple copies of SQLite linked into the same application"

Checkpoint + vacuum is probably not related.

On linux I can replicate the "recovered X frames from WAL file" warning from GEODIFF if from python I will start a connection, commit some changes (to create WAL file) and call createChangeset while the connection is still open - geodiff's internal sqlite thinks the other reader has closed incorrectly.

**Comment by wonder-sk** [comment]
Should be fixed now thanks to the upgrade to pygeodiff 1.0 which does not bundle sqlite internally.
